README.md

```diff
@@ -200,20 +202,96 @@ Retrieve a fully generated sitemap for your blog.
+**New in version 1.3.0**
+
+By the power of .NET generics, Content Fields can now be deserialized by the libary! The former method that would defer deserialization is still available to ease transition.
 
 #### RetrieveContentFields() Parameters:
 | Parameter|Description|
 | ---|---|
 |keys|String array of desired keys|
+|parameterDictionary(optional)|Dictionary of additional parameters, such as "locale" or "test"|
+
+#### RetrieveContentFields() Exceptions:
+| Exception|Description|
+| ---|---|
+|[ContentFieldObjectMismatchException](#contentfieldobjectmismatchexception)|This exception will be thrown when the library can't fit the returned data into the passed object class. |
 
 #### Examples:
 ```C#
 var keys = new string[2] { "team_members[name=Elon]", "homepage_headline" };
-var contentFields = await butterClient.RetrieveContentFieldsJSONAsync(keys);
+var dict = new Dictionary&lt;string, string&gt;()
+            {
+                { "locale", "de" },
+                { "test", "1" }
+            };
+var teamMembersAndHeadline = butterClient.RetrieveContentFields<TeamMembersHeadline>(keys, dict);
 
 ```
 
+**Legacy Content Fields JSON documentation** :
+
+As demonstrated in the [Content Fields documentation](https://buttercms.com/docs/api/#content-fields), any number of user-defined content fields can be retrieved from the API. Expanding on the examples in the docs, https://api.buttercms.com/v2/content/?keys=homepage_headline,team_members&auth_token=321478403e868f0fc41f0115731f330ff720ce0b returns an object that won't fit neatly in one C# object. For this reason, the client forwards the JSON string response and leaves deserialization up to the caller.
+
+#### RetrieveContentFieldsJSON() Parameters:
+| Parameter|Description|
+| ---|---|
+|keys|String array of desired keys|
+|parameterDictionary(optional)|Dictionary of additional parameters, such as "locale" and "test"|
+
+#### Examples:
+```C#
+var keys = new string[2] { "team_members[name=Elon]", "homepage_headline" };
+var dict = new Dictionary&lt;string, string&gt;()
+            {
+                { "locale", "de" },
+                { "test", "1" }
+            };
+var contentFields = await butterClient.RetrieveContentFieldsJSONAsync(keys, dict);
+
+```
+
+## Pages
+### List Pages
+
+Listing Pages returns a [PagesResponse](#pagesresponse-class) object
+
+#### ListPages() Parameters:
+| Parameter|Description|
+| ---|---|
+|pageType| Desired page type|
+|parameterDictionary| Dictionary of additional parameters|
+
+#### ListPages() Exceptions:
+| Exception|Description|
+| ---|---|
+|[PagesObjectMismatchException](#pagesobjectmismatchexception)|This exception will be thrown when the library can't fit the returned data into the passed object class. |
+
+#### Examples:
+```C#
+PagesResponse<ProductPage> productPages = butterClient.ListPages<ProductPage>("products");
```

Comment:
Please add `parameterDictionary` example here (similar to the content fields one above)

Reply:
Also, please show how `ProductPage` generic is defined. I'm not sure if Generics are a new feature or very well known?  

I want it to be super clear to people they define _any_ Page model they want; that they create 100% customizable data models.  My concern is people will glance at this and think that the SDK only returns `ProductPage`.